controllers/bpfman-agent/application-program_test.go

```diff
@@ -0,0 +1,308 @@
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func TestBpfApplicationControllerCreate(t *testing.T) {
+	var (
+		// global config
+		name         = "fakeAppProgram"
+		namespace    = "bpfman"
+		bytecodePath = "/tmp/hello.o"
+		fakeNode     = testutils.NewNode("fake-control-plane")
+		ctx          = context.TODO()
+		// fentry program config
+		bpfFentryFunctionName = "fentry_test"
+		fentryFunctionName    = "do_unlinkat"
+		fentryBpfProgName     = fmt.Sprintf("%s-%s-%s-%s", name, "fentry", fakeNode.Name, "do-unlinkat")
+		fentryBpfProg         = &bpfmaniov1alpha1.BpfProgram{}
+		fentryFakeUID         = "ef71d42c-aa21-48e8-a697-82391d801a81"
+		// kprobe program config
+		bpfKprobeFunctionName       = "kprobe_test"
+		kprobeFunctionName          = "try_to_wake_up"
+		kprobeBpfProgName           = fmt.Sprintf("%s-%s-%s-%s", name, "kprobe", fakeNode.Name, "try-to-wake-up")
+		kprobeBpfProg               = &bpfmaniov1alpha1.BpfProgram{}
+		kprobeFakeUID               = "ef71d42c-aa21-48e8-a697-82391d801a82"
+		kprobeOffset                = 0
+		kprobeRetprobe              = false
+		kprobecontainerpid    int32 = 0
+	)
+
+	// A AppProgram object with metadata and spec.
+	App := &bpfmaniov1alpha1.BpfApplication{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: bpfmaniov1alpha1.BpfApplicationSpec{
+			BpfAppCommon: bpfmaniov1alpha1.BpfAppCommon{
+				NodeSelector: metav1.LabelSelector{},
+				ByteCode: bpfmaniov1alpha1.BytecodeSelector{
+					Path: &bytecodePath,
+				},
+			},
+			Programs: []bpfmaniov1alpha1.BpfApplicationProgram{
+				{
+					Type: bpfmaniov1alpha1.ProgTypeFentry,
+					Fentry: &bpfmaniov1alpha1.FentryProgramInfo{
+						BpfProgramCommon: bpfmaniov1alpha1.BpfProgramCommon{
+							BpfFunctionName: bpfFentryFunctionName,
+						},
+						FunctionName: fentryFunctionName,
+					},
+				},
+				{
+					Type: bpfmaniov1alpha1.ProgTypeKprobe,
+					Kprobe: &bpfmaniov1alpha1.KprobeProgramInfo{
+						BpfProgramCommon: bpfmaniov1alpha1.BpfProgramCommon{
+							BpfFunctionName: bpfKprobeFunctionName,
+						},
+						FunctionName: kprobeFunctionName,
+						Offset:       uint64(kprobeOffset),
+						RetProbe:     kprobeRetprobe,
+					},
+				},
+			},
+		},
+	}
+
+	// Objects to track in the fake client.
+	objs := []runtime.Object{fakeNode, App}
+
+	// Register operator types with the runtime scheme.
+	s := scheme.Scheme
+	s.AddKnownTypes(bpfmaniov1alpha1.SchemeGroupVersion, App)
+	s.AddKnownTypes(bpfmaniov1alpha1.SchemeGroupVersion, &bpfmaniov1alpha1.BpfApplicationList{})
+	s.AddKnownTypes(bpfmaniov1alpha1.SchemeGroupVersion, &bpfmaniov1alpha1.BpfApplication{})
+	s.AddKnownTypes(bpfmaniov1alpha1.SchemeGroupVersion, &bpfmaniov1alpha1.BpfProgramList{})
+	s.AddKnownTypes(bpfmaniov1alpha1.SchemeGroupVersion, &bpfmaniov1alpha1.BpfProgram{})
+	s.AddKnownTypes(bpfmaniov1alpha1.SchemeGroupVersion, &bpfmaniov1alpha1.FentryProgramList{})
+	s.AddKnownTypes(bpfmaniov1alpha1.SchemeGroupVersion, &bpfmaniov1alpha1.KprobeProgramList{})
```

Comment:
You shouldn't need these since we don't actually create/list any Fentry and Kprobe entries within the k8s API right?

Reply:
yes removed